src/main/java/com/hivemq/client/mqtt/mqtt5/message/publish/Mqtt5Publish.java

```diff
@@ -110,6 +110,15 @@
      */
     @NotNull Mqtt5UserProperties getUserProperties();
 
+    /**
+     * Acknowledges this Publish message.
+     *
+     * @throws UnsupportedOperationException if manual acknowledgement is not enabled.
```

Comment:
Why not also if its a qos 0 message?

Reply:
This decision was made in the hivemq-mqtt-client slack channel.
The reason is that the user would need to check the QoS level. When subscribing with max QoS 2, you will get QoS 0, 1 and 2 messages. So it is easier for the user to just acknowledge after the business logic and depending on the QoS level different actions are performed (nothing for QoS 0)